x-pack/test/security_solution_api_integration/test_suites/detections_response/detection_engine/rule_execution_logic/trial_license_complete_tier/execution_logic/machine_learning_alert_suppression.ts

```diff
@@ -0,0 +1,1083 @@
+          const firstTimestamp = new Date().toISOString();
+          const firstAnomaly = {
+            ...baseAnomaly,
+            timestamp: firstTimestamp,
+          };
+          await indexListOfDocuments([firstAnomaly]);
+
+          const ruleWithoutSuppression = { ...ruleProps, alert_suppression: undefined };
+          const createdRule = await createRule(supertest, log, {
+            ...ruleWithoutSuppression,
+            from: firstTimestamp,
+          });
+          const alerts = await getAlerts(supertest, log, es, createdRule);
+
+          expect(alerts.hits.hits).toHaveLength(1);
+
+          // update the rule to include suppression
+          await patchRule(supertest, log, {
+            id: createdRule.id,
+            alert_suppression: ruleProps.alert_suppression,
+          });
+
+          const secondTimestamp = new Date().toISOString();
+          const secondAnomaly = {
+            ...baseAnomaly,
+            timestamp: secondTimestamp,
+          };
+
+          // Add more anomalies, then disable and re-enable to trigger another
+          // rule run. The second anomalies should create a new suppressed alert, since the original was not suppressed.
+          await indexListOfDocuments([secondAnomaly, secondAnomaly, secondAnomaly]);
+          await patchRule(supertest, log, { id: createdRule.id, enabled: false });
+          await patchRule(supertest, log, { id: createdRule.id, enabled: true });
+          const secondAlerts = await getOpenAlerts(
+            supertest,
+            log,
+            es,
+            createdRule,
+            RuleExecutionStatusEnum.succeeded,
+            undefined,
+            new Date()
+          );
+
+          expect(secondAlerts.hits.hits).toHaveLength(2);
+          // assert that the first alert does not have suppression fields
+          expect(secondAlerts.hits.hits[0]._source).toEqual(
+            expect.not.objectContaining({
+              [ALERT_SUPPRESSION_TERMS]: expect.anything(),
+              [ALERT_ORIGINAL_TIME]: expect.anything(),
+              [ALERT_SUPPRESSION_START]: expect.anything(),
+              [ALERT_SUPPRESSION_END]: expect.anything(),
+              [ALERT_SUPPRESSION_DOCS_COUNT]: expect.anything(),
+            })
+          );
+
+          expect(secondAlerts.hits.hits[1]._source).toEqual(
+            expect.objectContaining({
+              [ALERT_SUPPRESSION_TERMS]: [
+                {
+                  field: 'user.name',
+                  value: ['root'],
+                },
+              ],
+              [ALERT_ORIGINAL_TIME]: firstTimestamp,
+              [ALERT_SUPPRESSION_START]: firstTimestamp,
+              [ALERT_SUPPRESSION_END]: secondTimestamp,
+              [ALERT_SUPPRESSION_DOCS_COUNT]: 3, // TODO this means that the original anomaly was used as the suppression base, and the three new were suppressed into it (1 original + three new - 1 parent). Is this correct?
```

Comment:
@vitaliidm 👀

Reply:
This one is interesting.

Because generated id of alert with enabled suppression includes suppression terms in a list of hashed values - for every new suppression configuration it would result in a new alert. That would allow to use the same document as base alert for multiple suppression configuration, instead of its deduplication. 

So, it would create alert instead of deduplication. But would suppress the rest of potential alerts that match suppression configuration.

Originally, this behaviour was introduced in [custom query rule type](https://github.com/elastic/kibana/blob/8.13/x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/query/alert_suppression/wrap_suppressed_alerts.ts#L76).

cc: @marshallmain, do you remember more about this behaviour?

Reply:
I think the original reasoning for including the bucket terms in `_id` had to do with pre-aggregating the source documents in the search query for KQL rules so we don't have access to all source docs in the executor. For the original KQL implementation, we pre-aggregate as part of the search query so we can't easily identify and filter out duplicates within each bucket. If the first document in the bucket, that we use as the basis for the alert, is a duplicate of an existing alert, I didn't want to make more queries to figure out which documents in the bucket were duplicates because the buckets could have a huge number of docs, nor did I want to discard the bucket, so the solution was to include the bucket info in the `_id` to make it more unique and rely on query filters (`buildBucketHistoryFilter`) to remove duplicates with time range filters in the query stage for subsequent rule executions.

When we suppress alerts in memory we have all the candidate alerts already so filtering out duplicates by `_id` and grouping the rest together is no problem. Maybe an ideal solution would separate the `_id` from some kind of "suppression bucket ID" for in memory suppression so we can remove duplicates by `_id`, then group candidate alerts into buckets and compute each bucket ID based on the terms and values, then for each bucket either create a new alert or append to an existing alert with the same bucket ID?

Reply:
oh yeah `ALERT_INSTANCE_ID` already is the "suppression bucket ID" - so maybe for in memory suppression we just don't need the bucket info in `_id` at all?

Reply:
> oh yeah ALERT_INSTANCE_ID already is the "suppression bucket ID" - so maybe for in memory suppression we just don't need the bucket info in _id at all?

In this case, behaviour for the rest of rules on suppression would be different to query rule, since they won't have duplicate alerts that can be suppressed with different values. But since, they are in memory and their behaviour is different anyway, I think this should be fine

Reply:
@marshallmain , @rylnd FYI: https://github.com/elastic/kibana/pull/184453